src/utils/jobsTableUtils.ts

```diff
@@ -0,0 +1,191 @@
+import { Job, JobGroup, JobFilter, JobOrder } from "model"
+import { ColumnSpec } from "pages/JobsPage"
+import GetJobsService from "services/GetJobsService"
+import GroupJobsService from "services/GroupJobsService"
+
+export interface BaseJobTableRow {
+  rowId: string
+}
+
+export interface JobRow extends BaseJobTableRow {
+  // Job details
+  jobId?: string
+  jobSet?: string
+  queue?: string
+  state?: string
+  cpu?: number
+  memory?: string
+  ephemeralStorage?: string
+}
+
+export interface JobGroupRow extends BaseJobTableRow {
+  isGroup: true // The ReactTable version of this doesn't seem to play nice with manual/serverside expanding
+  count?: number
+  subRows?: JobTableRow[]
+
+  // Some subfield of JobRow that this row is grouped on
+  [groupedField: string]: unknown
+}
+
+export type JobTableRow = JobRow | JobGroupRow
+
+export const isJobGroupRow = (row: JobTableRow): row is JobGroupRow => "isGroup" in row
+
+function jobsToRows(jobs: Job[]): JobRow[] {
+  return jobs.map(
+    (job): JobRow => ({
+      rowId: "job:" + job.jobId,
+      jobId: job.jobId,
+      jobSet: job.jobSet,
+      queue: job.queue,
+      state: job.state,
+      cpu: job.cpu,
+      memory: job.memory,
+      ephemeralStorage: job.ephemeralStorage,
+    }),
+  )
+}
+
+function groupsToRows(groups: JobGroup[], baseRowId: string, groupingField: string): JobGroupRow[] {
+  return groups.map(
+    (group): JobGroupRow => ({
+      rowId: baseRowId + groupingField + ":" + group.name,
+      [groupingField]: group.name,
+
+      isGroup: true,
+      count: group.count,
+      subRows: [], // Will be set later if expanded
+    }),
+  )
+}
+
+interface FetchRowRequest {
+  parentRowId: string | undefined // Undefined means root rows
+  skip: number
+  take: number
+}
+interface FetchRowResult {
+  rows: JobRow[]
+  totalCount: number
+}
+const fetchRows = async (
```

Comment:
`rowRequest` is only ever a list with one element, so this can be simplified to only perform a single fetch

https://github.com/carlocamurri/lookout-v2/pull/3/files#diff-d73c28dbb46e86bbf0a72aa1e1380c592e817e525ff95bfbb8b6150959de94ffR97